### PR TITLE
Transaction waits for query() without returning a promise

### PR DESCRIPTION
**Pull request by dominykas**

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -187,18 +220,12 @@ describe("iw-mysql-wrapper", function () {
 						transactionScope.query("INSERT INTO `iw_mysql_wrapper_test` VALUES (1);"),
```
Comment (kestux) [inline test/index.test.js]: why this is still wrapped in Q.all?

File: test/index.test.js [hunk 2]
```diff
@@ -187,18 +220,12 @@ describe("iw-mysql-wrapper", function () {
...
 						transactionScope.query("BAD SQL;")
 					]);
```
Comment (kestux) [inline test/index.test.js:222]: Why transaction queries are still wrapped in Q.all?

File: test/index.test.js [hunk 3]
```diff
@@ -187,18 +220,12 @@ describe("iw-mysql-wrapper", function () {
...
-
-				}).catch(function verifyAfterRollback(err) {
+				}).then(neverCallMe("Query should throw a parse error.")).catch(function verifyAfterRollback(err) {
 
 					expect(err.code).to.eql("ER_PARSE_ERROR");
 					return Q.all([
-						savedScope.query("SELECT * FROM `iw_mysql_wrapper_test`").then(function (rows) {
-							expect(rows).to.eql([], "Transaction scope: should be rolled back");
+						savedScope.query("SELECT 1;").then(neverCallMe("Transaction should be closed.")).catch(function (err) {
+							expect(err.code).to.eql("E_TRANSACTION_CLOSED");
```
Comment (kestux) [inline test/index.test.js:228]: If I've understood it correct this assert has nothing to do with the roll back result. And it's not about the "one assert per test", but about the the test name should define what is expected. This assert is about closing the transaction after rolling back.

File: test/index.test.js [hunk 4]
```diff
@@ -172,6 +183,27 @@ describe("iw-mysql-wrapper", function () {
...
+						}),
+						db.query("SELECT * FROM `iw_mysql_wrapper_test`").then(function (rows) {
+							expect(rows).to.eql([{id: 1}, {id: 2}], "DB scope: should have values in the table (transaction is complete)");
+						})
+					]);
+				});
+
+			});
+
+			it("should execute multiple queries (without returning a promise)", function () {
```
Comment (kestux) [inline test/index.test.js:199]: Maybe something like "should execute multiple queries within the same promise" would define the test better?

File: test/index.test.js [hunk 5]
```diff
@@ -157,11 +165,14 @@ describe("iw-mysql-wrapper", function () {
 
 			it("should execute multiple queries", function () {
```
Comment (kestux) [inline test/index.test.js:166]: Maybe "should execute multiple queries and _close the transaction_"?

File: test/index.test.js (hunk 4 above)
Comment (dominykas) [inline test/index.test.js:199]: Not really.

File: index.js [hunk 6]
```diff
@@ -57,9 +57,23 @@ var Database = function (options) {
...
 
 	var executeTransaction = function (connection, inTransactionFn) {
+		var queryFn = getQueryFn(connection),
+			transactionComplete = false;
+
+		var transactionScope = {
+				query: function (query, args) {
+					if (transactionComplete) {
+						var error = new Error("Transaction is already closed");
+						error.code ="E_TRANSACTION_CLOSED";
```
Comment (kestux) [inline index.js:67]: space after `=`

File: test/index.test.js (hunk 3 above)
Comment (dominykas) [inline test/index.test.js:228]: There are several asserts in here:
1. That parse error was rethrown
2. That you can no longer execute queries inside the transaction
3. That the transaction was not committed whatsoever - i.e. the table is empty.

If you have a better idea how to assert that the transaction was rolled back - let me know.

Comment (dominykas) [inline test/index.test.js:228]: As for "test name should define what's expected" - what's expected is written out explicitly in the test code, we can argue to death about naming here, but I don't think that would give us any value, other than the fun of the discussion. Do you honestly think that the test name is problematic or is it pure dogma?

File: test/index.test.js (hunk 2 above)
Comment (dominykas) [inline test/index.test.js:222]: Because they were that way.

File: test/index.test.js (hunk 5 above)
Comment (dominykas) [inline test/index.test.js:166]: Maybe "it should be a transaction and behave like a transaction"...

**Pull request merged**